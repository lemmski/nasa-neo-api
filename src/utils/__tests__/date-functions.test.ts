import {
  eachMonthOfInterval,
  eachIntervalStartDateForMonth,
  eachIntervalEndDateForMonth,
} from "../date-functions";
describe("[date-functions.eachMonthOfInterval]", () => {
  it("should generate every month between start year and end year", async () => {
    const intervals = eachMonthOfInterval(
      new Date("2017-01-01"),
      new Date("2019-12-31")
    );
    expect(intervals).toMatchSnapshot();
    const sameYearIntervals = eachMonthOfInterval(
      new Date("2017-01-01"),
      new Date("2017-12-31")
    );
    expect(sameYearIntervals).toMatchSnapshot();
  });
});
describe("[date-functions.eachIntervalStartDateForMonth]", () => {
  it("should generate start dates for max 7 days intervals of the months for January 2017", async () => {
    const date = new Date("2017-01-01t13:00:00.000z");
    const januaryMonthsIntervalStartDates = eachIntervalStartDateForMonth(date);
    expect(januaryMonthsIntervalStartDates).toMatchSnapshot();
  });
  it("should generate start dates for max 7 days intervals of the months for February 2017", async () => {
    const date = new Date("2017-02-01t13:00:00.000z");
    const februaryMonthsIntervalStartDates = eachIntervalStartDateForMonth(
      date
    );
    expect(februaryMonthsIntervalStartDates).toMatchSnapshot();
  });
  it("should generate start dates for max 7 days intervals of the months for April 2017", async () => {
    const date = new Date("2017-04-01t13:00:00.000z");
    const aprilMonthsIntervalStartDates = eachIntervalStartDateForMonth(date);
    expect(aprilMonthsIntervalStartDates).toMatchSnapshot();
  });
  it("should generate start dates for max 7 days intervals of the months for February 2020 (leap year)", async () => {
    const date = new Date("2020-02-01t13:00:00.000z");
    const leapYearMonthsIntervalStartDates = eachIntervalStartDateForMonth(
      date
    );
    expect(leapYearMonthsIntervalStartDates).toMatchSnapshot();
  });
});
describe("[date-functions.eachIntervalEndDateForMonth]", () => {
  it("should generate end dates for max 7 days intervals of the months for January 2017", async () => {
    const januaryIntervalsWithStartAndEndDate = eachIntervalEndDateForMonth(
      eachIntervalStartDateForMonth(new Date("2017-01-01t13:00:00.000z"))
    );
    expect(januaryIntervalsWithStartAndEndDate).toMatchSnapshot();
  });
  it("should generate end dates for max 7 days intervals of the months for February 2017", async () => {
    const februaryIntervalsWithStartAndEndDate = eachIntervalEndDateForMonth(
      eachIntervalStartDateForMonth(new Date("2017-02-01t13:00:00.000z"))
    );
    expect(februaryIntervalsWithStartAndEndDate).toMatchSnapshot();
  });
  it("should generate end dates for max 7 days intervals of the months for April 2017", async () => {
    const aprilIntervalsWithStartAndEndDate = eachIntervalEndDateForMonth(
      eachIntervalStartDateForMonth(new Date("2017-04-01t13:00:00.000z"))
    );
    expect(aprilIntervalsWithStartAndEndDate).toMatchSnapshot();
  });
  it("should generate end dates for max 7 days intervals of the months for February 2020 (leap year)", async () => {
    const leapYearIntervalsWithStartAndEndDate = eachIntervalEndDateForMonth(
      eachIntervalStartDateForMonth(new Date("2020-02-01t13:00:00.000z"))
    );
    expect(leapYearIntervalsWithStartAndEndDate).toMatchSnapshot();
  });
});
