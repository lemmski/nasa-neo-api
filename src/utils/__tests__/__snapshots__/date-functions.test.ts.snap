// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[date-functions.eachIntervalEndDateForMonth] should generate end dates for max 7 days intervals of the months for April 2017 1`] = `
Array [
  Array [
    2017-04-01T13:00:00.000Z,
    2017-04-07T13:00:00.000Z,
  ],
  Array [
    2017-04-08T13:00:00.000Z,
    2017-04-14T13:00:00.000Z,
  ],
  Array [
    2017-04-15T13:00:00.000Z,
    2017-04-21T13:00:00.000Z,
  ],
  Array [
    2017-04-22T13:00:00.000Z,
    2017-04-28T13:00:00.000Z,
  ],
  Array [
    2017-04-29T13:00:00.000Z,
    2017-04-30T13:00:00.000Z,
  ],
]
`;

exports[`[date-functions.eachIntervalEndDateForMonth] should generate end dates for max 7 days intervals of the months for February 2017 1`] = `
Array [
  Array [
    2017-02-01T13:00:00.000Z,
    2017-02-07T13:00:00.000Z,
  ],
  Array [
    2017-02-08T13:00:00.000Z,
    2017-02-14T13:00:00.000Z,
  ],
  Array [
    2017-02-15T13:00:00.000Z,
    2017-02-21T13:00:00.000Z,
  ],
  Array [
    2017-02-22T13:00:00.000Z,
    2017-02-28T13:00:00.000Z,
  ],
]
`;

exports[`[date-functions.eachIntervalEndDateForMonth] should generate end dates for max 7 days intervals of the months for February 2020 (leap year) 1`] = `
Array [
  Array [
    2020-02-01T13:00:00.000Z,
    2020-02-07T13:00:00.000Z,
  ],
  Array [
    2020-02-08T13:00:00.000Z,
    2020-02-14T13:00:00.000Z,
  ],
  Array [
    2020-02-15T13:00:00.000Z,
    2020-02-21T13:00:00.000Z,
  ],
  Array [
    2020-02-22T13:00:00.000Z,
    2020-02-28T13:00:00.000Z,
  ],
  Array [
    2020-02-29T13:00:00.000Z,
    2020-02-29T13:00:00.000Z,
  ],
]
`;

exports[`[date-functions.eachIntervalEndDateForMonth] should generate end dates for max 7 days intervals of the months for January 2017 1`] = `
Array [
  Array [
    2017-01-01T13:00:00.000Z,
    2017-01-07T13:00:00.000Z,
  ],
  Array [
    2017-01-08T13:00:00.000Z,
    2017-01-14T13:00:00.000Z,
  ],
  Array [
    2017-01-15T13:00:00.000Z,
    2017-01-21T13:00:00.000Z,
  ],
  Array [
    2017-01-22T13:00:00.000Z,
    2017-01-28T13:00:00.000Z,
  ],
  Array [
    2017-01-29T13:00:00.000Z,
    2017-01-31T13:00:00.000Z,
  ],
]
`;

exports[`[date-functions.eachIntervalStartDateForMonth] should generate start dates for max 7 days intervals of the months for April 2017 1`] = `
Array [
  2017-04-01T13:00:00.000Z,
  2017-04-08T13:00:00.000Z,
  2017-04-15T13:00:00.000Z,
  2017-04-22T13:00:00.000Z,
  2017-04-29T13:00:00.000Z,
]
`;

exports[`[date-functions.eachIntervalStartDateForMonth] should generate start dates for max 7 days intervals of the months for February 2017 1`] = `
Array [
  2017-02-01T13:00:00.000Z,
  2017-02-08T13:00:00.000Z,
  2017-02-15T13:00:00.000Z,
  2017-02-22T13:00:00.000Z,
]
`;

exports[`[date-functions.eachIntervalStartDateForMonth] should generate start dates for max 7 days intervals of the months for February 2020 (leap year) 1`] = `
Array [
  2020-02-01T13:00:00.000Z,
  2020-02-08T13:00:00.000Z,
  2020-02-15T13:00:00.000Z,
  2020-02-22T13:00:00.000Z,
  2020-02-29T13:00:00.000Z,
]
`;

exports[`[date-functions.eachIntervalStartDateForMonth] should generate start dates for max 7 days intervals of the months for January 2017 1`] = `
Array [
  2017-01-01T13:00:00.000Z,
  2017-01-08T13:00:00.000Z,
  2017-01-15T13:00:00.000Z,
  2017-01-22T13:00:00.000Z,
  2017-01-29T13:00:00.000Z,
]
`;

exports[`[date-functions.eachMonthOfInterval] should generate every month between start year and end year 1`] = `
Array [
  2017-01-01T13:00:00.000Z,
  2017-02-01T13:00:00.000Z,
  2017-03-01T13:00:00.000Z,
  2017-04-01T12:00:00.000Z,
  2017-05-01T12:00:00.000Z,
  2017-06-01T12:00:00.000Z,
  2017-07-01T12:00:00.000Z,
  2017-08-01T12:00:00.000Z,
  2017-09-01T12:00:00.000Z,
  2017-10-01T12:00:00.000Z,
  2017-11-01T13:00:00.000Z,
  2017-12-01T13:00:00.000Z,
  2018-01-01T13:00:00.000Z,
  2018-02-01T13:00:00.000Z,
  2018-03-01T13:00:00.000Z,
  2018-04-01T12:00:00.000Z,
  2018-05-01T12:00:00.000Z,
  2018-06-01T12:00:00.000Z,
  2018-07-01T12:00:00.000Z,
  2018-08-01T12:00:00.000Z,
  2018-09-01T12:00:00.000Z,
  2018-10-01T12:00:00.000Z,
  2018-11-01T13:00:00.000Z,
  2018-12-01T13:00:00.000Z,
  2019-01-01T13:00:00.000Z,
  2019-02-01T13:00:00.000Z,
  2019-03-01T13:00:00.000Z,
  2019-04-01T12:00:00.000Z,
  2019-05-01T12:00:00.000Z,
  2019-06-01T12:00:00.000Z,
  2019-07-01T12:00:00.000Z,
  2019-08-01T12:00:00.000Z,
  2019-09-01T12:00:00.000Z,
  2019-10-01T12:00:00.000Z,
  2019-11-01T13:00:00.000Z,
  2019-12-01T13:00:00.000Z,
]
`;

exports[`[date-functions.eachMonthOfInterval] should generate every month between start year and end year 2`] = `
Array [
  2017-01-01T13:00:00.000Z,
  2017-02-01T13:00:00.000Z,
  2017-03-01T13:00:00.000Z,
  2017-04-01T12:00:00.000Z,
  2017-05-01T12:00:00.000Z,
  2017-06-01T12:00:00.000Z,
  2017-07-01T12:00:00.000Z,
  2017-08-01T12:00:00.000Z,
  2017-09-01T12:00:00.000Z,
  2017-10-01T12:00:00.000Z,
  2017-11-01T13:00:00.000Z,
  2017-12-01T13:00:00.000Z,
]
`;
